### PR TITLE
[5.0] Add django.db.models.GeneratedField

### PR DESCRIPTION
**Pull request by palfrey**
# I have made things!

## Related issues

Part of https://github.com/typeddjango/django-stubs/issues/1493 changes

### CODE REVIEW

File: django-stubs/db/models/fields/generated.pyi [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from typing import Any
+
+from django.db import models
+from django.db.backends.base.base import BaseDatabaseWrapper
+from django.db.models.expressions import Expression
+from django.utils.datastructures import DictWrapper
+
+class GeneratedField(models.Field):
+    def __init__(
+        self, *, expression: Expression, output_field: models.Field, db_persist: bool | None = ..., **kwargs: Any
```
Comment (Viicos) [inline django-stubs/db/models/fields/generated.pyi]: I think it's better to add all the possible arguments, as it is done for other `Field` subclasses

Comment (palfrey) [reply django-stubs/db/models/fields/generated.pyi]: Done

File: django-stubs/db/models/fields/generated.pyi [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+from django.db import models
+from django.db.backends.base.base import BaseDatabaseWrapper
+from django.db.models.expressions import Expression
+from django.db.models.fields import _ErrorMessagesMapping, _FieldChoices
+from django.db.models.sql import Query
+from django.utils.datastructures import DictWrapper
+from django.utils.functional import _StrOrPromise
+
+class GeneratedField(models.Field):
+    generated: Literal[True] = True
```
Comment (sobolevn) [inline django-stubs/db/models/fields/generated.pyi]: We don't need default values here

**Review by sobolevn — COMMENTED** [review]
Why do we ignore `cached_col`?

**Comment by palfrey** [comment]
> Why do we ignore `cached_col`?

Hit the error at the top of that ignore block `# Ignore @cached_property error "cannot reconcile @property on stub with runtime object"`

File: django-stubs/db/models/fields/generated.pyi (hunk 2 above)
Comment (palfrey) [reply django-stubs/db/models/fields/generated.pyi]: Removed

File: django-stubs/db/models/fields/generated.pyi [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        unique_for_year: str | None = ...,
+        choices: _FieldChoices | None = ...,
+        help_text: _StrOrPromise = ...,
+        db_column: str | None = ...,
+        db_comment: str | None = ...,
+        db_tablespace: str | None = ...,
+        validators: Iterable[_ValidatorCallable] = ...,
+        error_messages: _ErrorMessagesMapping | None = ...,
+        **kwargs: Any,
+    ): ...
```
Comment (sobolevn) [inline django-stubs/db/models/fields/generated.pyi]:
```suggestion
    ) -> None: ...
```

File: django-stubs/db/models/fields/generated.pyi [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+from django.db.models.fields import _ErrorMessagesMapping, _FieldChoices
+from django.db.models.sql import Query
+from django.utils.datastructures import DictWrapper
+from django.utils.functional import _StrOrPromise
+
+class GeneratedField(models.Field):
+    generated: Literal[True] = ...
+    db_returning: Literal[True] = ...
+    _query: Query | None = ...
+    output_field: models.Field | None = ...
```
Comment (sobolevn) [inline django-stubs/db/models/fields/generated.pyi]:
```suggestion
    generated: Literal[True]
    db_returning: Literal[True]
    _query: Query | None
    output_field: models.Field | None
```

**Review by sobolevn — APPROVED** [review]

**Pull request merged**